sidecar_acceptor_test.go

```diff
@@ -32,67 +31,46 @@ var (
 	}
 )
 
-func newTestDB(t *testing.T) (*clientdb.DB, func()) {
-	tempDir, err := ioutil.TempDir("", "client-db")
-	require.NoError(t, err)
-
-	db, err := clientdb.New(tempDir, clientdb.DBFilename)
-	if err != nil {
-		require.NoError(t, os.RemoveAll(tempDir))
-		t.Fatalf("unable to create new db: %v", err)
-	}
-
-	return db, func() {
-		require.NoError(t, db.Close())
-		require.NoError(t, os.RemoveAll(tempDir))
-	}
-}
-
 // TestRegisterSidecar makes sure that registering a sidecar ticket verifies the
 // offer signature contained within, adds the recipient node's information to
 // the ticket and stores it to the local database.
 func TestRegisterSidecar(t *testing.T) {
-	t.Parallel()
 
-	mockSigner := test.NewMockSigner()
-	mockWallet := test.NewMockWalletKit()
-	mockSigner.Signature = testOfferSig.Serialize()
-
-	acceptor := NewSidecarAcceptor(&SidecarAcceptorConfig{
-		SidecarDB:  nil,
-		AcctDB:     nil,
-		Signer:     mockSigner,
-		Wallet:     mockWallet,
-		NodePubKey: ourNodePubKey,
-		ClientCfg:  auctioneer.Config{},
-	})
+	emptySetter := func(
+		ticket *sidecar.Ticket, sc *test.MockSignerClient,
+		wc *test.MockWalletKitClient, store *sidecar.MockStore,
+	) {
+	}
 
-	existingTicket, err := sidecar.NewTicket(
-		sidecar.VersionDefault, 1_000_000, 200_000, 2016,
-		providerPubKey, false,
-	)
-	require.NoError(t, err)
+	emptyCheck := func(t *testing.T, ticket *sidecar.Ticket) {
+	}
 
 	testCases := []struct {
 		name        string
 		ticket      *sidecar.Ticket
 		expectedErr string
-		check       func(t *testing.T, ticket *sidecar.Ticket)
+		mockSetter  func(
+			ticket *sidecar.Ticket, sc *test.MockSignerClient,
+			wc *test.MockWalletKitClient, store *sidecar.MockStore,
+		)
+		check func(t *testing.T, ticket *sidecar.Ticket)
 	}{{
-		name: "ticket with ID exists",
+		name:        "unable to register sidecar if ticket does not have a valid state",
+		ticket:      &sidecar.Ticket{},
+		expectedErr: "ticket is in invalid state",
+		mockSetter:  emptySetter,
+		check:       emptyCheck,
+	}, {
+		name: "unable to register sidecar if signature is missing",
 		ticket: &sidecar.Ticket{
-			ID:    existingTicket.ID,
 			State: sidecar.StateOffered,
-			Offer: sidecar.Offer{
-				SignPubKey:     providerPubKey,
-				SigOfferDigest: testOfferSig,
-			},
 		},
-		expectedErr: "ticket with ID",
+		expectedErr: "error verifying sidecar offer",
```

Comment:
We should actually check that the error contains `offer in ticket is not signed` here.

Reply:
yes, better be as specific as we can and if we change the message we change the test than to take the "higher level" msg (Y)